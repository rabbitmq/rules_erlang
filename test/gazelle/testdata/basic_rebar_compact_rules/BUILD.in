# gazelle:erlang_generate_fewer_bytecode_rules
