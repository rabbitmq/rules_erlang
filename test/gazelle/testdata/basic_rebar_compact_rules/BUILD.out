load("@rules_erlang//:erlang_bytecode2.bzl", "erlang_bytecode", "erlc_opts")
load("@rules_erlang//:erlang_app.bzl", "erlang_app", "test_erlang_app")
load("@rules_erlang//:xref2.bzl", "xref")
load("@rules_erlang//:dialyze.bzl", "dialyze", "plt")
load("@rules_erlang//:eunit2.bzl", "eunit")
load("@rules_erlang//:ct.bzl", "assert_suites2")

# gazelle:erlang_generate_fewer_bytecode_rules

erlc_opts(
    name = "erlc_opts",
    values = select({
        "@rules_erlang//:debug_build": [
            "+debug_info",
            "+recv_opt_info",
            "+warn_export_vars",
            "+warn_obsolete_guard",
            "+warn_shadow_vars",
        ],
        "//conditions:default": [
            "+debug_info",
            "+deterministic",
            "+recv_opt_info",
            "+warn_export_vars",
            "+warn_obsolete_guard",
            "+warn_shadow_vars",
        ],
    }),
    visibility = [":__subpackages__"],
)

erlc_opts(
    name = "test_erlc_opts",
    values = select({
        "@rules_erlang//:debug_build": [
            "+debug_info",
            "-DTEST=1",
        ],
        "//conditions:default": [
            "+debug_info",
            "+deterministic",
            "-DTEST=1",
        ],
    }),
    visibility = [":__subpackages__"],
)

plt(
    name = "base_plt",
    visibility = [":__subpackages__"],
)

erlang_bytecode(
    name = "other_beam",
    srcs = glob(["src/**/*.erl"]),
    hdrs = [":public_and_private_hdrs"],
    app_name = "seshat",
    dest = "ebin",
    erlc_opts = "//:erlc_opts",
)

erlang_bytecode(
    name = "test_other_beam",
    testonly = True,
    srcs = glob(["src/**/*.erl"]),
    hdrs = [":public_and_private_hdrs"],
    app_name = "seshat",
    dest = "test",
    erlc_opts = "//:test_erlc_opts",
)

filegroup(
    name = "beam_files",
    srcs = [":other_beam"],
)

filegroup(
    name = "test_beam_files",
    testonly = True,
    srcs = [":test_other_beam"],
)

erlang_bytecode(
    name = "test_seshat_counters_server_test_beam",
    testonly = True,
    srcs = ["test/seshat_counters_server_test.erl"],
    outs = ["test/seshat_counters_server_test.beam"],
    app_name = "seshat",
    erlc_opts = "//:test_erlc_opts",
)

erlang_bytecode(
    name = "test_seshat_test_beam",
    testonly = True,
    srcs = ["test/seshat_test.erl"],
    outs = ["test/seshat_test.beam"],
    app_name = "seshat",
    erlc_opts = "//:test_erlc_opts",
)

filegroup(
    name = "srcs",
    srcs = glob([
        "src/**/*.app.src",
        "src/**/*.erl",
    ]),
)

filegroup(
    name = "private_hdrs",
    srcs = glob(["src/**/*.hrl"]),
)

filegroup(
    name = "public_hdrs",
    srcs = glob(["include/**/*.hrl"]),
)

filegroup(
    name = "priv",
    srcs = glob(["priv/**/*"]),
)

filegroup(
    name = "license_files",
    srcs = glob(["LICENSE*"]),
)

filegroup(
    name = "public_and_private_hdrs",
    srcs = [
        ":private_hdrs",
        ":public_hdrs",
    ],
)

filegroup(
    name = "all_srcs",
    srcs = [
        ":public_and_private_hdrs",
        ":srcs",
    ],
)

erlang_app(
    name = "erlang_app",
    srcs = [":all_srcs"],
    hdrs = [":public_hdrs"],
    app_name = "seshat",
    beam_files = [":beam_files"],
    extra_apps = [
        "crypto",
        "sasl",
    ],
    license_files = [":license_files"],
    priv = [":priv"],
)

alias(
    name = "seshat",
    actual = ":erlang_app",
    visibility = ["//visibility:public"],
)

test_erlang_app(
    name = "test_erlang_app",
    srcs = [":all_srcs"],
    hdrs = [":public_and_private_hdrs"],
    app_name = "seshat",
    beam_files = [":test_beam_files"],
    extra_apps = [
        "crypto",
        "sasl",
    ],
    license_files = [":license_files"],
    priv = [":priv"],
)

xref(
    name = "xref",
    target = ":erlang_app",
)

plt(
    name = "deps_plt",
    for_target = ":erlang_app",
    plt = "//:base_plt",
)

dialyze(
    name = "dialyze",
    plt = ":deps_plt",
    target = ":erlang_app",
)

eunit(
    name = "eunit",
    compiled_suites = [
        ":test_seshat_counters_server_test_beam",
        ":test_seshat_test_beam",
    ],
    target = ":test_erlang_app",
)

assert_suites2()
